core/src/modules/cloud.ts

```diff
@@ -87,19 +90,42 @@ export class CloudClient {
       process.env.GROUPAROO_CLOUD_API_URL ?? "https://cloud.grouparoo.com";
   }
 
-  async request(url: string, options?: RequestInit) {
+  async request(
+    url: string,
+    options?: RequestInit & { _buildFormData?: () => FormData },
+    attempts = 0
+  ) {
     const fetchUrl = new URL(url, this.baseUrl);
     fetchUrl.searchParams.append("apiToken", this.token);
 
-    const res = await fetch(fetchUrl.toString(), options);
-    const data = await res.json();
-
-    if (res.status !== 200) {
-      if (data.error) throw new CloudError(data.error);
-      throw new Error(await res.text());
+    // when retrying a form based on a readStream, we need to restart the stream on every attempt
+    if (typeof options?._buildFormData === "function") {
+      const formData = options._buildFormData.bind(this)();
+      //@ts-ignore typings are incorrectly not allowing FormData to be set as body
+      options.body = formData;
+      options.headers = formData.getHeaders();
     }
```

Comment:
@pedroslopez this pattern is kind of crazy - Since we are using a readStream to create the formData, we actually need to restart the stream for each upload.

Reply:
oh wow, makes sense